openai_assistant/utils/event_handler.py

```diff
@@ -149,8 +149,20 @@ async def process_thread_message(
         elements = adapter.get_elements()
 
         # Update the message content with the new text and elements
-        message.content = content
-        message.elements = elements
+        message.content = content + '\n'
+        last_source = ''
+        for element in elements:
```

Comment:
This needs to be moved to the get_elements code. Since they're no longer elements, we can rename it to get-citations, and simplify this code so that we can do do `content = await adapter.main()`